tests/e2e/testcases/localnetwork/suite.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package localnetwork
+
+import (
+	"github.com/ava-labs/avalanche-cli/tests/e2e/commands"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+var _ = ginkgo.Describe("[Local Network]", ginkgo.Ordered, func() {
+	ginkgo.AfterEach(func() {
+		commands.CleanNetwork()
+	})
+
+	ginkgo.It("can get correct status of local network", func() {
+		// no network started
+		out, err := commands.GetNetworkStatus()
+		gomega.Expect(err).ShouldNot(gomega.BeNil())
+		gomega.Expect(out).Should(gomega.ContainSubstring("network is not running"))
+
+		// start local network
+		out = commands.StartNetwork()
+		gomega.Expect(out).Should(gomega.ContainSubstring("Network ready to use"))
+
+		// check network status after starting
+		out, err = commands.GetNetworkStatus()
+		gomega.Expect(err).Should(gomega.BeNil())
+		gomega.Expect(out).Should(gomega.ContainSubstring("Network is Up"))
+
+		// Stop the network
+		commands.StopNetwork()
+
+		// check network status after stopping
+		out, err = commands.GetNetworkStatus()
+		gomega.Expect(err).ShouldNot(gomega.BeNil())
+		gomega.Expect(out).Should(gomega.ContainSubstring("network is not running"))
+	})
+
+	ginkgo.It("can start then stop a started network", func() {
+		out := commands.StartNetwork()
+		gomega.Expect(out).Should(gomega.ContainSubstring("Network ready to use"))
+
+		commands.StopNetwork()
+
+		out, err := commands.GetNetworkStatus()
+		gomega.Expect(err).ShouldNot(gomega.BeNil())
+		gomega.Expect(out).Should(gomega.ContainSubstring("network is not running"))
+	})
+
+	ginkgo.It("can start then stop a started network with --dont-save", func() {
+		out := commands.StartNetwork()
+		gomega.Expect(out).Should(gomega.ContainSubstring("Network ready to use"))
+
+		commands.StopNetwork("--dont-save")
```

Comment:
this should be a constant on cmd/flags/ used also in cmd/ code

Reply:
--dont-save does something on the state, can this be checked? (really it does not create a snapshot)
so the cases for snapshots are:
- stop without args: create a default snapshot
- stop with --dont-save: dont create a default snapshot
- stop with --snapshot-name: create a snapshot with a different name
network clean beforehand will remove the default snapshot, the other snapshot need to be removed by hand 
(this is order to check it is created)

Reply:
in e2e we are using the same `~/.avalanche-cli/snapshots` directory as user's daily CLI use. so e2e here may interfere with user's daily CLI uses, particularly the `default` snapshot. (btw what i need to do in this e2e here is to check existence and then removal) @felipemadero

Reply:
mmm your are totally right. options:
1) I am not opposite to temporal renaming of the default snapshot and then renaming it back again. testing is mainly
a dev team operation and we are expected to recover the default in case the "renaming it back again" part fails.
2) I am not opposite to have the snapshot dir being redefinable at the app level, so we on testing use a non production one.
as you wish (probably 2 is best no?)

Reply:
Yes #2 sounds better indeed. btw do we have our CLI options categorized as public and internal using? to add such a new option, do we want to make it public to users too?

Reply:
Our public options are set at config file level, and there is not structure regarding public vs
internal options. Either is it public in the config file, or is internal in various places, and usually fixed.
But we don't have any public option to move something out of `~/.avalanche-cli`. 
I believe that atm, for less impact (eg in support), we can just keep it private, to continue 
having that centralization.

Reply:
Btw, interested in ideas related to public vs internal options. Feel free to propose ideas on prototype PRs

Reply:
@felipemadero please check this: https://github.com/ava-labs/avalanche-cli/pull/2761

Reply:
@sukantoraymond made a valid point here: https://github.com/ava-labs/avalanche-cli/pull/2761#issuecomment-2825545133 given that there's already `IsE2E()` uses in code, that's the easiest change to use a separate e2e directory for testing purposes, which is also the least risky. 
i think we can do this `IsE2E` solution for now and leave public\internal CLI arguments for more discussions

@felipemadero

Reply:
I don't agree with having separate special cmd flows or settings for e2e inside the code. It is happening rn I now, but that seems tech debt to me. Maybe we can use it as a solution, but have a deeper discussion on this.
Probably we want to be able to configure the CLI dir by flag, and use that everywhere in all E2E commands. 
But not using a hidden flag but a public one. Anyway, feel free to use E2E env var and continue thinking if
we want to do better.